src/plugins/controls/public/control_group/component/control_group_component.tsx

```diff
@@ -254,16 +299,48 @@ export const ControlGroup = () => {
                 </DragOverlay>
               </DndContext>
             </EuiFlexItem>
-            {showAddButton && (
-              <EuiFlexItem grow={false}>
-                <EuiButtonIcon
-                  size="s"
-                  iconSize="m"
-                  display="base"
-                  iconType={'plusInCircle'}
-                  aria-label={ControlGroupStrings.management.getAddControlTitle()}
-                  onClick={() => controlGroup.openAddDataControlFlyout()}
-                />
+            {showAppendedButtonGroup && (
+              <EuiFlexItem
+                grow={false}
+                className="controlGroup--endButtonGroup"
+                data-test-subj="controlGroup--endButtonGroup"
+              >
+                <EuiFlexGroup responsive={false} gutterSize="s" alignItems="center">
+                  {showAddButton && (
+                    <EuiFlexItem grow={false}>
+                      <EuiToolTip content={ControlGroupStrings.management.getAddControlTitle()}>
+                        <EuiButtonIcon
+                          size="m"
+                          iconSize="m"
+                          display="base"
+                          iconType={'plusInCircle'}
+                          data-test-subj="controlGroup--addControlButton"
+                          aria-label={ControlGroupStrings.management.getAddControlTitle()}
+                          onClick={() => controlGroup.openAddDataControlFlyout()}
+                        />
+                      </EuiToolTip>
+                    </EuiFlexItem>
+                  )}
+                  {showApplySelections && (
+                    <EuiFlexItem grow={false}>
+                      <EuiButtonIcon
+                        size="m"
+                        disabled={!applyButtonEnabled}
+                        iconSize="m"
+                        display="fill"
+                        color={'success'}
+                        iconType={'check'}
+                        data-test-subj="controlGroup--applyFiltersButton"
+                        aria-label={ControlGroupStrings.management.getApplyButtonTitle(
+                          applyButtonEnabled
+                        )}
+                        onClick={() => {
+                          if (unpublishedFilters) controlGroup.publishFilters(unpublishedFilters);
+                        }}
+                      />
+                    </EuiFlexItem>
```

Comment:
I used to have a tooltip around the apply button that would change its text depending on if the button was enabled or not. However, there is a [known bug](https://github.com/elastic/eui/issues/6488) with React where the tooltip would get completely stuck when the button changes from enabled to disabled:

https://github.com/elastic/kibana/assets/8698078/d31e4110-e0ff-45a8-a046-4262b6486985

As a result of this, I've decided to remove the tooltip for now. In theory, we could have a tooltip in **either** the enabled or disabled state, but **not** both; I wasn't sure if this was